source/joker.c

```diff
@@ -248,68 +279,86 @@ void joker_object_shake(JokerObject *joker_object, mm_word sound_id)
-        int cursorPosX = fx2int(joker_object->sprite_object->x) + 8; // Offset of 16 pixels to center the text on the card
-        if (joker_effect.chips > 0)
-        {
-            char score_buffer[INT_MAX_DIGITS + 2]; // For '+' and null terminator
-            tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
-            tte_set_special(0xD000); // Blue
-            snprintf(score_buffer, sizeof(score_buffer), "+%d", joker_effect.chips);
-            tte_write(score_buffer);
-            cursorPosX += joker_score_display_offset_px;
-        }
-        if (joker_effect.mult > 0)
-        {
-            char score_buffer[INT_MAX_DIGITS + 2];
-            tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
-            tte_set_special(0xE000); // Red
-            snprintf(score_buffer, sizeof(score_buffer), "+%d", joker_effect.mult);
-            tte_write(score_buffer);
-            cursorPosX += joker_score_display_offset_px;
-        }
-        if (joker_effect.xmult > 0)
-        {
-            char score_buffer[INT_MAX_DIGITS + 2];
-            tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
-            tte_set_special(0xE000); // Red
-            snprintf(score_buffer, sizeof(score_buffer), "X%d", joker_effect.xmult);
-            tte_write(score_buffer);
-            cursorPosX += joker_score_display_offset_px;
-        }
-        if (joker_effect.money > 0)
-        {
-            char score_buffer[INT_MAX_DIGITS + 2];
-            tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
-            tte_set_special(0xC000); // Yellow
-            snprintf(score_buffer, sizeof(score_buffer), "+%d", joker_effect.money);
-            tte_write(score_buffer);
-            cursorPosX += joker_score_display_offset_px;
-        }
+    // Use the custom message to show a retrigger
+    if (joker_effect.retrigger) {
+        snprintf(joker_effect.message, 7, "Again!");
+    }
 
-        joker_object->joker->processed = true; // Mark the joker as processed
-        joker_object_shake(joker_object, SFX_CARD_SELECT); // TODO: Add a sound effect for scoring the joker
+    const int joker_score_display_offset_px = (MAX_CARD_SCORE_STR_LEN + 1)*TTE_CHAR_SIZE;
+    // + 1 For space
 
-        return true;
+    int cursorPosX = fx2int(joker_object->sprite_object->x) + 8; // Offset of 16 pixels to center the text on the card
+    if (joker_effect.chips > 0)
+    {
+        char score_buffer[INT_MAX_DIGITS + 2]; // For '+' and null terminator
+        tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
+        // TODO fix magic number
+        tte_set_special(TTE_BLUE_PB * 4096); // Blue
+        snprintf(score_buffer, sizeof(score_buffer), "+%d", joker_effect.chips);
+        tte_write(score_buffer);
+        cursorPosX += joker_score_display_offset_px;
+    }
+    if (joker_effect.mult > 0)
+    {
+        char score_buffer[INT_MAX_DIGITS + 2];
+        tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
+        // TODO fix magic number
+        tte_set_special(TTE_RED_PB * 4096); // Red
```

Comment:
```suggestion
        tte_set_special(TTE_RED_PB * TTE_SPECIAL_PB_MULT_OFFSET);
```

Batch - Add TTE_SPECIAL_PB_MULT_OFFSET